docs/bin/plugin_formatter.py

```diff
@@ -159,6 +147,22 @@ def rst_xline(width, char="="):
     return char * width
 
 
+def documented_type(text):
+    ''' Convert any python type to a type for documentation '''
+
+    if isinstance(text, Undefined):
+        return '-'
+    if text == 'str':
+        return 'string'
```

Comment:
If we start expanding type names, we should have a sanity checks which prevents people from using the long names in `type:` of their documentation. (IMO)

Reply:
That's already in place, you can't just invent new types.

Reply:
Ok, great!

Reply:
FYI this is enforced by:
https://github.com/ansible/ansible/blob/625a2d14700a4939334b385be0e72d39c83a1da2/test/sanity/validate-modules/schema.py#L132

Reply:
I know, I have another PR #50010 that cleans up the Return Values bits (more consistency).